src/backend/distributed/utils/reference_table_utils.c

```diff
@@ -133,56 +133,48 @@ ReplicateAllReferenceTablesToNode(char *nodeName, int nodePort)
 	ListCell *referenceTableCell = NULL;
 	ListCell *referenceShardIntervalCell = NULL;
 	List *workerNodeList = ActivePrimaryNodeList();
-	uint32 workerCount = 0;
-	Oid firstReferenceTableId = InvalidOid;
-	uint32 referenceTableColocationId = INVALID_COLOCATION_ID;
+	int workerCount = list_length(workerNodeList);
 
-	/* if there is no reference table, we do not need to do anything */
-	if (list_length(referenceTableList) == 0)
+	/* if there is no reference table, we do not need to replicate anything */
+	if (list_length(referenceTableList) > 0)
 	{
-		return;
-	}
-
-	/*
-	 * We sort the reference table list to prevent deadlocks in concurrent
-	 * ReplicateAllReferenceTablesToAllNodes calls.
-	 */
-	referenceTableList = SortList(referenceTableList, CompareOids);
-	foreach(referenceTableCell, referenceTableList)
-	{
-		Oid referenceTableId = lfirst_oid(referenceTableCell);
-		List *shardIntervalList = LoadShardIntervalList(referenceTableId);
-		ShardInterval *shardInterval = (ShardInterval *) linitial(shardIntervalList);
+		/*
```

Comment:
Minor: we could move declarations of `referenceTableList/referenceShardIntervalList/referenceShardIntervalCell/referenceTableCell` in to the if block.